inmoose/cohort_qc/cohort_metric.py

```diff
@@ -0,0 +1,864 @@
+            ax=ax,
+            x="Dataset",
+            y="Expression",
+            hue="Type",
+            data=data_after_melted,
+            palette={"Mixed": "red", "Non-Mixed": "grey"},
+        )
+        ax.set_title("Gene expression distribution after correction")
+        ax.set_xlabel("Dataset (Batch)")
+        ax.set_ylabel("Gene Expression")
+        ax.tick_params(axis="x", rotation=45)
+        ax.legend(title="Dataset Type", loc="upper right")
+
+        # Return the plot as a figure object
+        fig.tight_layout()
+        return fig
+
+    def compare_sample_distribution_total(self) -> plt.Figure:
+        """
+        Compare the overall distribution of samples by dataset before and after correction across all covariate combinations.
+
+        Returns
+        -------
+        matplotlib.figure.Figure
+            The generated plot object.
+        """
+        self.clinical_df["Covariate_Combination"] = (
+            self.clinical_df[self.covariates].astype(str).agg("-".join, axis=1)
+        )
+
+        data_after = self.data_expression_df.copy()
+        mean_after = data_after.mean(axis=0).reset_index()
+        mean_after.columns = ["Sample", "Mean_Expression"]
+        mean_after["Covariate_Combination"] = self.clinical_df.loc[
+            mean_after["Sample"], "Covariate_Combination"
+        ].values
+
+        width = (11 / (12 - min(11, mean_after["Covariate_Combination"].nunique()))) + 4
+        heigth = max(
+            10,
+            (
+                max(
+                    len(item)
+                    for item in mean_after["Covariate_Combination"].unique()
+                    if isinstance(item, str)
+                )
+            )
+            / 3,
+        )
+
+        if self.data_expression_df_before is not None:
+            data_before = self.data_expression_df_before.copy()
+            mean_before = data_before.mean(axis=0).reset_index()
+            mean_before.columns = ["Sample", "Mean_Expression"]
+            mean_before["Covariate_Combination"] = self.clinical_df.loc[
+                mean_before["Sample"], "Covariate_Combination"
+            ].values
+
+            fig, axs = plt.subplots(nrows=2, ncols=1, figsize=(width, heigth))
+
+            # Plot before correction
+            sns.violinplot(
+                ax=axs[0],
+                x="Covariate_Combination",
+                y="Mean_Expression",
+                data=mean_before,
+            )
+            axs[0].set_title(
+                "Mean of sample gene expression distribution before correction"
+            )
+            axs[0].set_xlabel("Covariate combination")
+            axs[0].set_ylabel("Mean gene expression by sample")
+            axs[0].tick_params(axis="x", rotation=45)
+            ax = axs[1]
+        else:
+            fig, ax = plt.subplots(nrows=1, ncols=1, figsize=(width, heigth / 2))
+
+        # Plot after correction
+        sns.violinplot(
+            ax=ax,
```

Comment:
Same here, how does it work when ax=axs?

Reply:
The code line ax = axs[1] when we have to plot before and after allow to use the second part of the subplot